force-app/main/default/classes/SOQL.cls

```diff
@@ -232,24 +292,30 @@ public with sharing class SOQL {
      */
     public static Map<Id, String> getAccountIdsToNamesNotInCountry(String country) {
         // Create a map to hold the Account Ids to Account Names
-        Map<Id, String> accountIdsToNames = new Map<Id, String>();
+        Map<Id, String> accountMap = new Map<Id, String>();
 
         // Query the Accounts
-        // Loop through each Account and add the Id and Name to the map
-
-        return null; // Replace null with the variable you used to store the result
+      List <Account> myAccountList = [SELECT ID, Name FROM Account WHERE BillingCountry != :country];
+      // Loop through each Account and add the Id and Name to the map
+      for (Account acc : myAccountList){
+        accountMap.put(acc.ID, acc.Name);
+      }
+      return accountMap;
     }
+ //       return null; // Replace null with the variable you used to store the result
+  //  }
 
     /**
-     * Question 19
+     * Question 19 
      * Retrieves the Account Name for a given Contact.
      * Example: getAccountNameForContact('003XXXXXXXXXXXX') should return the Account Name for the Contact with the Id
      * @param contactId The Id of the Contact.
      * @return The Account Name of the Contact.
      */
     public static String getAccountNameForContact(String contactId) {
-        return null; // Replace null with the variable you used to store the result
-    }    
+    Account acc = [SELECT Name FROM Account WHERE Id IN (SELECT AccountId FROM Contact WHERE Id = :contactId) LIMIT 1];
+    return acc != null ? acc.Name : null;
```

Comment:
I'd recommend reversing this query. Query the contact and traverse the relationships to get the account name